Python/optimizer_bytecodes.c

```diff
@@ -117,6 +117,10 @@ dummy_func(void) {
 
     op(_GUARD_TYPE_VERSION, (type_version/2, owner -- owner)) {
         assert(type_version);
+        if (sym_is_type_subclass(owner)) {
```

Comment:
I don't think this should be necessary.
If this case isn't handled correctly by `sym_matches_type_version` and `sym_set_type_version` below, then those need fixing.

Reply:
We need this because it might have a type version that matches, but is a type subclass, not a normal object.

Reply:
`_GUARD_TYPE_VERSION` guards the version number of `owner`'s class. Whether `owner` is a class or not seems irrelevant.